### DISCUSSION TITLE
Dexcom One with a serial number starting with C

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2624] [Show and tell]
xDrip supports them starting with release January 12, 2023: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.01.12

**Comment by paravoid** [comment]
Thank you @jamorham for your work on this. Deeply grateful for your work on xDrip in general but also specifically here, to build something that works and restores functionality or provides more options for the community, and for building this in a short amount of time, and releasing it gratis. 

With that said and the hope I won't sound too ungrateful, I'd like to express my disappointment with the fact that this was introduced as a binary plugin and came with infrastructure to load more of these binary plugins.

The community has suffered by the plethora of commercial, proprietary and undocumented protocols in the broader ecosystem. Proprietary tech that is attached to one's body, and transmits metrics about ourselves or our loved ones (or in the case of pumps, receives critical, potentially lethal treatments for) with no ability to inspect, understand how it works, or give us the ability to use our data, about ourselves in any way we wish.

Tremendous amount of effort has gone into the reverse engineering of many of those handshakes and protocols to get to where we are today. xDrip has been a light in the darkness of this purposefully uninteroperable world, filled with carefully engineered walled gardens. Its offer of free and open source reimplementation of many of these handshakes and protocols, exposing data in a myriad of ways (HTTP, broadcast intents etc.) has enabled the rest of the ecosystem to flourish. Recent trends (iCGMs, Android's Health Platform API, or even your work on companion mode) gave some hope that this may become even easier in the future.

Interoperability is not at stake with this plugin functionality as I understand it, and especially given you seemingly seem to be open to other projects using this plugin freely. However, after years of fighting with the proprietary ecosystems, seeing xDrip compromising with binary, proprietary plugins felt like a reversal on its core ideals and goals, and was, quite honestly, troubling to see.

I'd like to respectfully ask you to reconsider. I understand that this may be about funding and that you've got to pay the bills somehow, but I'd like to ask you to consider exploring avenues such as crowdfunding (GitHub sponsors, Patreon, Ko-fi, Kickstarter etc.). I know myself and many in the community would be happy to pay, either as gratitude for your past efforts, or bounty-style, like goals to add support for new hardware. Many /are/ paying already for hardware like Anubis or Bubble/MiaoMiao etc. to workaround interoperability constraints of their own devices already.

Again, I wish I'm not coming across as demanding. I'm extremely thankful for all of your work, and you don't really owe us anything. I just wish you could find other ways to do this, potentially the help of the community you've been instrumental in building :)

**Comment by jamorham** [comment]
@paravoid Thanks for your kind words. In the plugin documentation readme file I described that the plan is to release it open source in due course. That is how I would prefer things to be from the outset, but I am operating within various constraints... What I have tried to do here is provide the best solution I can today so that people are able to get access to their data with these transmitters and continue to enjoy those benefits.

In the many years I have been working on xDrip, I have not asked for any financial donations and have provided everything for free.  I have steered the project in the way which I think best helps users, always with a view to core principles and improving lives through the use of technology.  I am also a professional software engineer by trade and have often taken on commercial projects that I have been approached about. Many of which are closely related to the types of things xDrip does because I am well known as having a lot of expertise in those areas.  I think this commercial activity helps xDrip overall and in this case an existing project has rapidly allowed me to solve a particular issue for users. 

I'm not really sure how a sponsor/bounty type model could work and it seems like it maybe could create its own problems. I'm not closed to the idea but I'm also not convinced how helpful it would be.

**Reply by paravoid** [reply]
@jamorham Seems like you were good on your promise ;) Thanks so much for today's release!
